src/main/java/com/google/sps/algorithms/MatchRanking.java

```diff
@@ -0,0 +1,121 @@
+package com.google.sps.algorithms;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class MatchRanking {
+
+    /**
+    * Function takes collection of user's saved bios, all user bios, and new match bios and 
+    * returns a list of the new match bios in ranked order from highest to lowest score 
+    * (higher score = more likely that the user will select it). 
+    */
+    public static List<String> rankMatches(Collection<String> savedMatchBios, Collection<String> allUserBios, Collection<String> newMatchBios) {
+        Map<String, Double> newMatchScores = scoreNewMatches(savedMatchBios, allUserBios, newMatchBios);
+        return sortBiosByScore(newMatchScores);
+    }
+
+    /**
+    * Function that takes the three bio collections and returns a map with each new user bio mapped to it's score.
+    */
+    private static Map<String, Double> scoreNewMatches(Collection<String> savedMatchBios, Collection<String> allUserBios, Collection<String> newMatchBios) {
+        //count words in all of the saved matches
+        Map<String, Integer> savedMatchesWordCount = countWordInstances(savedMatchBios);
+
+        //count words in all of the user bios
+        Map<String, Integer> allUserWordCount = countWordInstances(allUserBios);
+
+        //calculate score for each new match bio
+        Map<String, Double> newBioScores = new HashMap<String, Double>();
+        for(String newBio : newMatchBios) {
+            double bioScore = calculateBioScore(newBio, savedMatchesWordCount, allUserWordCount);
+            newBioScores.put(newBio, new Double(bioScore));
+        }
+
+        return newBioScores;
+    }
+
+    /**
+    * Function that takes the map of bios and their scores and returns a list of the bios in sorted order from highest to lowest score.
```

Comment:
Once I integrate this into the actual application, I am going to call MatchRanking from MatchQuery with the bio strings, then probably move the sort method to MatchQuery to sort the bios that it returns. I will probably also keep it in MatchRanking because it is useful for testing.